### PR TITLE
fix(agent-templates): repair escaped-emoji text from builder generations

### PR DESCRIPTION
**Pull request by saulmc**
## Summary

A builder generation shipped an agent (dev template `3f6023de-e986-472b-a54a-23a02563f99a`, "Skyler") whose `emoji` field holds the literal 18-character text `\ud83c\udf24\ufe0f` instead of 🌤️ — rendered verbatim in the app. The prompt body of the same row carries matching debris, including broken lone surrogates (`Skyler \ud83c`), so the model wrote emoji as `\uXXXX` escape *text*; on the wire the backslashes were JSON-escaped, `JSON.parse` faithfully returned literal backslash text, and nothing downstream validates the field (`json_schema` only enforces `type: "string"`).

## Fix

- `decodeEmojiEscapes`: decodes literal surrogate-pair escape text and emoji-sequence escapes (VS15/VS16, ZWJ, keycap), and drops lone-surrogate halves. Applied to `agentName`, `description`, and `prompt` in `parseTemplateResponse`.
- `sanitizeEmojiField`: decode + require at least one emoji-class character (`Extended_Pictographic`, regional indicator, or keycap), else `""`. Applied to the emoji field on all three producer paths: main generation, GitHub passthrough (`📦` fallback), and pasted-content classifier (`🤖` fallback).
- Passthrough prompt content is intentionally not decoded — it's user/repo-authored, where literal `\uXXXX` text can be legitimate.

Existing rows written before this fix (e.g. Skyler) keep the corrupt value; that one is broken beyond the emoji ("Let me restart this properly" mid-prompt) and is worth regenerating rather than patching.

## Testing

- `pnpm vitest run tests/template-gen.service.test.ts` — 51/51 pass (5 new tests cover decode, lone-surrogate debris, field gating, and an end-to-end double-escaped wire payload).
- `pnpm typecheck`, `pnpm format:check`, `pnpm eslint` on touched files — clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/xmtplabs/codesmith/convos-backend/pr/308"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783882994&installation_id=128169237&pr_number=308&repository=xmtplabs%2Fconvos-backend&return_to=https%3A%2F%2Fgithub.com%2Fxmtplabs%2Fconvos-backend%2Fpull%2F308&signature=6aec8dc9e16a6a427b050a3447538c4acc8ae74ff2d20fc3a052aa9614b307f5"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

> [!NOTE]
> ### Fix escaped-emoji text in agent template builder generations
> - Adds `decodeEmojiEscapes` to convert literal Unicode escape sequences (e.g. `\uD83C\uDF24`) from model outputs into real glyphs, handling surrogate pairs, ZWJ sequences, and keycap combiners.
> - Adds `sanitizeEmojiField` to validate that an emoji field contains at least one actual emoji character, returning an empty string otherwise.
> - Applies these helpers in `parseTemplateResponse` so `agentName`, `description`, and `prompt` no longer contain raw escape text, and `emoji` is either a valid emoji or blank.
> - Applies `sanitizeEmojiField` in `tryGithubPassthrough` and `tryContentPassthrough` so malformed emoji values fall back to their defaults (`📦` and `🤖`).
> - Behavioral Change: agent templates that previously surfaced literal escape debris (e.g. `\uD83C\uDF24`) in text fields will now show decoded glyphs or empty strings.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 75dea4b.</sup>
> 
>

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed emoji field handling in agent template generation to properly decode escaped emoji sequences and validate characters. Invalid or malformed emoji data is now sanitized for consistency.

* **Tests**
  * Added comprehensive unit tests for emoji decoding and sanitization behavior, including edge cases and validation scenarios.

### CODE REVIEW

File: src/api/v2/agent-templates/services/templateGen.ts [hunk 1]
```diff
@@ -1587,6 +1587,42 @@ export async function generateTemplate(
...
+export function decodeEmojiEscapes(text: string): string {
+  return text
+    .replace(/\\u(d[89ab][0-9a-f]{2})\\u(d[c-f][0-9a-f]{2})/gi, (_, hi, lo) =>
+      String.fromCharCode(parseInt(hi, 16), parseInt(lo, 16)),
+    )
+    .replace(/\\u(fe0e|fe0f|200d|20e3)/gi, (_, hex) =>
+      String.fromCharCode(parseInt(hex, 16)),
+    )
+    .replace(/\\ud[89a-f][0-9a-f]{2}/gi, "");
+}
```
[the remark this answers is not in the document]
Comment (saulmc) [reply src/api/v2/agent-templates/services/templateGen.ts:1624]: Expanded the JSDoc in 75dea4b — purpose, the sequences handled (surrogate pairs, VS15/VS16, ZWJ, keycap), lone-surrogate removal, and pass-through behavior. Kept the package's prose-JSDoc style (no `@param`/`@returns` tags are used anywhere in agent-templates).

File: src/api/v2/agent-templates/services/templateGen.ts [hunk 2]
```diff
@@ -1587,6 +1587,42 @@ export async function generateTemplate(
...
+ *  one emoji-class character. Anything else (prose, escape debris) becomes ""
+ *  so a non-emoji string can never ship in the field. */
+export function sanitizeEmojiField(emoji: string): string {
+  const decoded = decodeEmojiEscapes(emoji).trim();
+  return /\p{Extended_Pictographic}|\p{Regional_Indicator}|\u20e3/u.test(
+    decoded,
+  )
+    ? decoded
+    : "";
+}
```
[the remark this answers is not in the document]
Comment (saulmc) [reply src/api/v2/agent-templates/services/templateGen.ts:1637]: Expanded the JSDoc in 75dea4b — repair + gate semantics, the emoji-class criteria, and the empty-string contract. Also extracted the validation regex to a named `EMOJI_CLASS_REGEX` constant. Prose-JSDoc style per package convention.
